EDDI/MainWindow.xaml.cs

```diff
@@ -858,50 +858,32 @@ public static void prepareLog(IProgress<string> progress)
                 string issueLogFile = issueLogDir + "eddi_issue.log";
                 string desktopPath = Environment.GetFolderPath(Environment.SpecialFolder.Desktop) + @"\eddi_issue.zip";
 
-                string[] log;
+                // Create a temporary issue log file, delete any remnants from prior issue reporting
                 lock (logLock)
                 {
-                    log = File.ReadAllLines(Constants.DATA_DIR + @"\eddi.log");
+                    Directory.CreateDirectory(issueLogDir);
+                    File.WriteAllText(issueLogFile, File.ReadAllText(Constants.DATA_DIR + @"\eddi.log"));
                 }
-                if (log.Length == 0) { return; }
-
-                progress.Report("");
-                List<string> outputLines = new List<string>();
-                // Use regex to isolate DateTimes from the string
-                Regex recentLogsRegex = new Regex(@"^(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2})");
 
-                foreach (string line in log)
+                // Truncate log files more than the specified size MB in size
+                long maxLogSizeMB = 2;
+                long maxLogSizeBytes = maxLogSizeMB * (long)Math.Pow(1024, 2);
+                FileInfo logFile = new FileInfo(issueLogFile);
+                if (logFile.Length > maxLogSizeBytes) 
                 {
-                    try
+                    using (MemoryStream ms = new MemoryStream((int)maxLogSizeBytes))
                     {
-                        // Parse log file lines so that we can examine DateTimes
-                        string linedatestring = recentLogsRegex.Match(line).Value;
-                        string formatString = "s"; // i.e. DateTimeFormatInfo.SortableDateTimePattern
-                        if (DateTime.TryParseExact(linedatestring, formatString, CultureInfo.InvariantCulture, DateTimeStyles.AssumeUniversal, out DateTime linedate))
+                        using (FileStream issueLog = new FileStream(issueLogFile, FileMode.Open, FileAccess.ReadWrite))
                         {
-                            linedate = linedate.ToUniversalTime();
-                            double elapsedHours = (DateTime.UtcNow - linedate).TotalHours;
-                            // Fill the issue log with log lines from the most recent hour only
-                            if (elapsedHours < 1.0)
-                            {
-                                outputLines.Add(line);
-                            }
+                            issueLog.Seek(-1 * maxLogSizeBytes, SeekOrigin.End);
+                            issueLog.CopyTo(ms);
```

Comment:
Should probably seek to after the next line break.

Reply:
What's the concern exactly? I'm not sure I follow.

Reply:
My concern is breaking the log in the middle of a line.